File: expedia-server/booking-server/src/main/java/org/clonecoder/bookingserver/domain/BaseEntity.java
package org.clonecoder.bookingserver.domain;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.MappedSuperclass;
import java.time.LocalDateTime;

@MappedSuperclass
@Getter
@Setter
public class BaseEntity {
    private String lastModifiedBy;
    private LocalDateTime lastModified;
    private String createdBy;
    private LocalDateTime createdAt;
}

File: platform-server/member-server/src/main/kotlin/org/clonecoder/member/domain/BaseEntity.kt
package org.clonecoder.member.domain

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import org.springframework.data.annotation.CreatedBy
import org.springframework.data.annotation.LastModifiedBy
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.ZonedDateTime
import javax.persistence.EntityListeners
import javax.persistence.MappedSuperclass

@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
abstract class BaseEntity {
    @CreationTimestamp
    var createdAt: ZonedDateTime = ZonedDateTime.now()
    @CreatedBy
    var createdBy: String = SYSTEM_PROPERTY_AUTHOR
    @UpdateTimestamp
    var lastModifiedAt: ZonedDateTime = ZonedDateTime.now()
    @LastModifiedBy
    var lastModifiedBy: String = SYSTEM_PROPERTY_AUTHOR
}

private const val SYSTEM_PROPERTY_AUTHOR = "system"